frontend/public/components/factory/table.tsx

```diff
@@ -460,73 +335,31 @@ export type RowFunctionArgs<T = any, C = any> = {
 
 export type RowFunction<T = any, C = any> = (args: RowFunctionArgs<T, C>) => React.ReactElement;
 
-export type VirtualBodyProps = {
-  customData?: any;
+export type VirtualBodyProps<D = any, C = any> = {
+  customData?: C;
   Row: RowFunction;
   height: number;
   isScrolling: boolean;
-  onChildScroll: (...args) => any;
-  data: any[];
+  onChildScroll: (params: Scroll) => void;
+  data: D[];
   columns: any[];
   scrollTop: number;
   width: number;
   expand: boolean;
 };
 
-export type TableProps = {
-  customData?: any;
-  customSorts?: { [key: string]: any };
-  data?: any[];
-  defaultSortFunc?: string;
-  defaultSortField?: string;
-  defaultSortOrder?: SortByDirection;
-  showNamespaceOverride?: boolean;
-  filters?: { [key: string]: any };
-  Header: (...args) => any[];
-  loadError?: string | Object;
-  Row?: RowFunction;
-  Rows?: (...args) => any[];
-  'aria-label': string;
-  onSelect?: OnSelect;
-  virtualize?: boolean;
-  NoDataEmptyMsg?: React.ComponentType<{}>;
-  EmptyMsg?: React.ComponentType<{}>;
-  loaded?: boolean;
-  reduxID?: string;
-  reduxIDs?: string[];
-  rowFilters?: any[];
-  label?: string;
-  columnManagementID?: string;
-  isPinned?: (val: any) => boolean;
-  staticFilters?: any[];
-  activeColumns?: Set<string>;
-  kinds?: string[];
-};
-
-type TablePropsFromState = {};
-
-type TablePropsFromDispatch = {};
-
-type TableOptionProps = {
-  UI: any;
-};
-
-type ComponentProps = {
-  data?: any[];
-  filters?: Object;
-  selected?: any;
-  match?: any;
-  kindObj?: K8sResourceKindReference;
-};
+type HeaderFunc = (componentProps: ComponentProps, t: TFunction) => any[];
```

Comment:
We shouldn't pass `t` as an argument. It's better to simply use `i18next.t` inside the function. (Not all `t`s are the same since `useTranslation` has params that control things like the default namespace.)

This applies here and a few other places it looks like.

Reply:
ok, thank you @spadgett . I removed every occurence of passing `t`